include/irr/asset/IImageAssetHandlerBase.h

```diff
@@ -31,6 +37,77 @@ class IImageAssetHandlerBase : public virtual core::IReferenceCounted
 			return width;
 		}
 
+		static inline core::vector3du32_SIMD calcPitchInBlocks(uint32_t width, uint32_t height, uint32_t depth, uint32_t blockByteSize)
+		{
+			constexpr auto VALUE_FOR_ALIGNMENT = 1;
+			core::vector3du32_SIMD retVal;
+			retVal.X = calcPitchInBlocks(width, blockByteSize);
+			retVal.Y = calcPitchInBlocks(height, VALUE_FOR_ALIGNMENT);
+			retVal.Z = calcPitchInBlocks(depth, VALUE_FOR_ALIGNMENT);
+			return retVal;
+		}
+
+		/*
+			Patch for not supported by OpenGL R8_SRGB formats.
+			Returns converted image. Input image needs to have
+			all the regions filled and texel buffer as well.
+
+			@devshgraphicsprogramming it should support layers and mipmaps
+			as well for GLI, do we have a filter that can handle it?
```

Comment:
every filter can handle multiple layers no problem.

As for mipmaps, just iterate through the mipmap chain and fire-off the filter once per mip map level